### ISSUE TITLE
Possible bug in MetadataRepository\Filter\CompositeFilter::toExpression

### ISSUE THREAD

**Issue by phavekes** [OpenConext/OpenConext-engineblock#1460]
[This issue is imported from pivotal - Originaly created at Feb 6, 2018 by Michiel Kodde](https://www.pivotaltracker.com/story/show/154961794)

**Backgrounds**
The `toExpression` method of:  `OpenConext\EngineBlock\Metadata\MetadataRepository\Filter\CompositeFilter::toExpression` Shows some strange behaviour. My IDE\'s inspector picked it up as a possible bug. On closer inspection this method does return unpredictable results.

The method is small enough to include it in this description:

```php
    public function toExpression($repositoryClassName)
    {
        $expressions = array();

        foreach ($this->filters as $filter) {
            $expression = $filter->toExpression($repositoryClassName);

            if (!$expression) {
                continue;
            }

            $expressions[] = $expression;
        }

        if (count($expression) === 0) {
            return null;
        }

        if (count($expression) === 1) {
            return $expression;
        }

        return new CompositeExpression(CompositeExpression::TYPE_AND, $expressions);
    }
```

The two if statements count the `expression` variable where `expressions` should be tested. Or at least that\'s what I would expect..

**Comment by phavekes** [comment]
Fixed in [EB#485](https://github.com/OpenConext/OpenConext-engineblock/pull/485) (Michiel Kodde - Feb 7, 2018)

**Comment by phavekes** [comment]
No longer an issue anyway since the fourfold implementation of each filter has been simplified to just one implementation for the database backend (and one implementation for non-behat tests that intentionally don\'t use the database), (Joris Steyn - Feb 20, 2018)
